modules/Bio/EnsEMBL/RNAProduct.pm

```diff
@@ -0,0 +1,976 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+
+=head1 CONTACT
+
+  Please email comments or questions to the public Ensembl
+  developers list at <http://lists.ensembl.org/mailman/listinfo/dev>.
+
+  Questions may also be sent to the Ensembl help desk at
+  <http://www.ensembl.org/Help/Contact>.
+
+=cut
+
+=head1 NAME
+
+Bio::EnsEMBL::RNAProduct - A class representing the mature RNA product
+of a transcript
+
+=head1 DESCRIPTION
+
+TODO
+
+=head1 SYNOPSIS
+
+  my $rnaproduct = Bio::EnsEMBL::RNAProduct->new(
+    -SEQ_START => 36,
+    -SEQ_END   => 58
+  );
+
+  # Stable-ID setter
+  $rnaproduct->stable_id('ENSM00090210');
+
+  # Get start and end position in the precursor transcript
+  my $start = $rnaproduct->start();
+  my $end = $rnaproduct->end();
+
+=cut
+
+
+package Bio::EnsEMBL::RNAProduct;
+
+use vars qw($AUTOLOAD);
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Utils::Exception qw(throw warning );
+use Bio::EnsEMBL::Utils::Argument qw( rearrange );
+use Bio::EnsEMBL::Utils::RNAProductTypeMapper;
+use Bio::EnsEMBL::Utils::Scalar qw( assert_ref wrap_array );
+use Scalar::Util qw(weaken);
+
+use Bio::EnsEMBL::Storable;
+
+use parent qw(Bio::EnsEMBL::Storable);
+
+
+=head2 new
+
+  Arg [-SEQ_START]    : The offset in the Transcript indicating the start
+                        position of the product sequence.
+  Arg [-SEQ_END]      : The offset in the Transcript indicating the end
+                        position of the product sequence.
+  Arg [-START_EXON]   : The Exon object in which the rnaproduct starts
+  Arg [-END_EXON]     : The Exon object in which the rnaproduct ends
+  Arg [-STABLE_ID]    : The stable identifier for this RNAPRoduct
+  Arg [-VERSION]      : The version of the stable identifier
+  Arg [-DBID]         : The internal identifier of this RNAProduct
+  Arg [-ADAPTOR]      : The RNAProductAdaptor for this RNAProduct
+  Arg [-SEQ]          : Manually sets the nucleotide sequence of this
+                        rnaproduct. May be useful if this rnaproduct is not
+                        stored in a database.
+  Arg [-CREATED_DATE] : the date the rnaproduct was created
```

Comment:
rnaproduct => RNA product
or should it be $rnaproduct or RNAProduct as used in previous pods?